src/System.CommandLine/IdentifierSymbol.cs

```diff
@@ -42,19 +42,18 @@ protected IdentifierSymbol(string name, string? description = null)
         /// <inheritdoc/>
         public override string Name
         {
-            get => _specifiedName ??= DefaultName;
             set
             {
-                if (_specifiedName is null || !string.Equals(_specifiedName, value, StringComparison.Ordinal))
+                if (_name is null || !string.Equals(_name, value, StringComparison.Ordinal))
                 {
                     AddAlias(value);
 
-                    if (_specifiedName is { })
+                    if (_name is { })
```

Comment:
Does this checks if _name is not null?
```suggestion
                    if (_name != null)
```

Reply:
It does and I also believe that it's very unintuitive way of checking for `null` ;)